cwl/src/main/scala/wdl4s/cwl/CwlCodecs.scala

```diff
@@ -10,18 +10,51 @@ import cats.syntax.either._
 import eu.timepit.refined.string._
 import io.circe.refined._
 import io.circe.literal._
+import cats.syntax.traverse._
+import cats.instances.list._
+import cats.instances.either._
+import cats.syntax.validated._
+
+import lenthall.validation.ErrorOr.ErrorOr
 
 object CwlCodecs {
 
-  def decodeCwl: Yaml => Either[Error, Cwl] = {
+  type EitherA[A] = Either[Error, A]
+
+  /**
+    * Parse a possibly top-level CWL file and return representations of this file and any referenced files.  This is
+    * very simplistic logic that assumes only one level of depth max.
+    * @return a `Map[String, CwlFile]` of filenames to `CwlFile`s.
+    */
+  def decodeCwl(yaml: String): ErrorOr[(CwlFile, Map[String, CwlFile])] = {
+    decodeSingleFileCwl(yaml) match {
+      case Right(clt: CommandLineTool) => (clt, Map.empty[String, CwlFile]).validNel
+      case Right(wf: Workflow) =>
+        val fileNames: List[String] = wf.steps.toList.flatMap(_.run.select[String].toList)
+
+        val fileNameToFiles: EitherA[List[(String, CwlFile)]] = fileNames.traverse[EitherA, (String, CwlFile)] {
+          fileName =>
+            val yaml = scala.io.Source.fromFile(fileName).getLines.mkString("\n")
```

Comment:
ToL: It looks like circe has a second `parse` method that takes a `java.io.Reader`, it could be interesting in the interest of memory to see if we can pass an input stream instead of a String